### PR TITLE
Fix issues when MONGODB_AUTH env var is set to true

### PR DESCRIPTION
**Pull request by lilian-pouliquen**
When setting `MONGODB_AUTH` to `true` the container stopped with an error due to the command used to create the users for the authentication: the `--fork` arg need to be used along with `--logpath <path>` or `--syslog`.
(Issue #31)

So I fixed it by adding the following env vars:

- `MONGODB_LOGPATH`: string, indicates the path to write the logs
- `MONGODB_SYSLOG`: bool default to `false`, activates the `--syslog` arg

These two vars are not set by default to let the user choose if they want to let the logs in standard output or not. So to fix the issue even if the vars are not set, the command which launches the local server to create users uses `--logpath ${MONGODB_LOGPATH:-$MONGODB_DBPATH/mongod.log} --logappend`.
It now should not fail.

I also fixed other issues I encountered while fixing this.
Then I updated the other scripts to match with the new fix.

I left some comment to explain why I changed some lines. Feel free to delete them.

I hope it will be helpful for you!

(EDIT: added reference to issue)

### CODE REVIEW

**Comment by tboerger** [comment]
That's a bigger pr I need to review. This will currently take some time.

**Comment by lilian-pouliquen** [comment]
@tboerger, no problem. If you need any explanation about my changes, feel free to ask me!

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
...
+[![Docker Build](https://github.com/dockhippie/mongodb/workflows/docker/badge.svg)](https://github.com/dockhippie/mongodb/actions?query=workflow%3Adocker) [![Readme Build](https://github.com/dockhippie/mongodb/workflows/readme/badge.svg)](https://github.com/dockhippie/mongodb/actions?query=workflow%3Areadme) [![Docker Size](https://img.shields.io/docker/image-size/webhippie/mongodb/latest)](https://github.com/dockhippie/mongodb) [![Docker Pulls](https://img.shields.io/docker/pulls/webhippie/mongodb)](https://hub.docker.com/r/webhippie/mongodb) [![GitHub Repo](https://img.shields.io/badge/github-repo-yellowgreen)](https://github.com/dockhippie/mongodb)
```
Comment (tboerger) [inline README.md:3]: This should be the same link as for `Docker Pulls`, e.g. https://hub.docker.com/r/webhippie/mongodb

File: README.md [hunk 2]
```diff
@@ -21,104 +21,108 @@ For the available versions please look at [Docker Hub](https://hub.docker.com/r/
...
+| `MONGODB_WIRED_TIGER_COLLECTION_BLOCK_COMPRESSOR`    | string |                                   | X    | X                |
+| `MONGODB_WIRED_TIGER_DIRECTORY_FOR_INDEXES`          | string |                                   | X    | X                |
+| `MONGODB_WIRED_TIGER_INDEX_PREFIX_COMPRESSION`       | string |                                   | X    | X                |
+| `MONGODB_WIRED_TIGER_JOURNAL_COMPRESSOR`             | string |                                   | X    | X                |
+| `MONGODB_WIRED_TIGER_MAX_CACHE_OVERFLOW_FILESIZE`    | string |                                   | X    | X                |
+| `MONGODB_ZSTD_DEFAULT_COMPRESSION_LEVEL`             | int    |                                   |      | X                |
 
 ## Inherited environment variables
 
-*  [webhippie/ubuntu](https://github.com/dockhippie/ubuntu#available-environment-variables)
```
Comment (tboerger) [inline README.md:113]: This space had been intentional, that's something Codacy complains about ;)

File: README.md [hunk 3]
```diff
@@ -21,104 +21,108 @@ For the available versions please look at [Docker Hub](https://hub.docker.com/r/
...
+* [webhippie/ubuntu](https://github.com/dockhippie/ubuntu#available-environment-variables)
 
 ## Contributing
 
 Fork -> Patch -> Push -> Pull Request
 
 ## Authors
 
-*  [Thomas Boerger](https://github.com/tboerger)
```
Comment (tboerger) [inline README.md:121]: This space had been intentional, that's something Codacy complains about ;)

**Review by tboerger — COMMENTED** [review]
At least I have started with some initial comments and resolved the conflict after merging a fix for the auth env variable :)

**Comment by lilian-pouliquen** [comment]
> At least I have started with some initial comments and resolved the conflict after merging a fix for the auth env variable :)

okay I read your comments, do you want me to change the things you pointed out? I let you the permissions to make changes to the code in the pull request. As I am new to this, I do not know well how it works :)

**Comment by tboerger** [comment]
Thanks for the contribution, I have cherry-picked your commits and added my personal note on top at https://github.com/dockhippie/mongodb/pull/37

**Pull request closed without merging**